add opennextjs streaming tests (#5993)
cloudflare/workerd: src/workerd/api/tests/opennextjs/README.md

# OpenNext.js SSR Test

> This directory contains tests for running actual OpenNext.js Cloudflare bundled output in workerd.

Reviewer: Updating 2 places only, there are more across the PR

```suggestion
# OpenNext SSR Test

This directory contains tests for running actual OpenNext Cloudflare bundled output in workerd.
```

The test verifies that workerd can correctly execute Next.js SSR applications built with the
`@opennextjs/cloudflare` adapter.

## Files

- `opennext-ssr-test.js` - Test cases for API routes, SSR pages, streaming, RSC, etc.
- `opennext-ssr-test.wd-test` - workerd test configuration
- `opennext-ssr-worker.js` - Bundled OpenNext.js worker (generated)

## Current Versions

- Next.js: 16.1.6
- @opennextjs/cloudflare: 1.16.1
- React: 19.x

## Regenerating the Bundled Worker

To regenerate `opennext-ssr-worker.js` with updated versions:

### 1. Create a new Next.js project

```bash
mkdir /tmp/nextjs-opennext-test && cd /tmp/nextjs-opennext-test
npm init -y
npm install next@latest react@latest react-dom@latest @opennextjs/cloudflare@latest
```

### 2. Create the project structure

**package.json** - Add scripts:

```json
{
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "opennext": "opennextjs-cloudflare build"
  }
}
```

**next.config.ts**:

```typescript
import type { NextConfig } from 'next';

const nextConfig: NextConfig = {
  // Disable image optimization (requires external service)
  images: {
    unoptimized: true,

>   },

Reviewer: not needed 
```suggestion
```

`@opennextjs/cloudflare` adapter.

## Files

- `opennext-ssr-test.js` - Test cases for API routes, SSR pages, streaming, RSC, etc.
- `opennext-ssr-test.wd-test` - workerd test configuration
- `src/` - Next.js application source files (JavaScript/JSX)

## Running the Test

The worker is automatically generated from source before the test runs:

```bash
bazel test //src/workerd/api/tests/opennextjs:opennext-ssr-test@
```

**Note:** This test is Linux-only (`target_compatible_with = ["@platforms//os:linux"]`) due to
the reliance on the Next.js/OpenNext build toolchain.

## How It Works

1. **opennextjs-build**: Runs `@opennextjs/cloudflare build` to compile the Next.js app and
   generate the OpenNext worker in `.open-next/`
2. **opennextjs-worker**: Runs `wrangler deploy --dry-run --outdir=dist` to bundle the worker
3. The output `dist/worker.js` is copied to `opennext-ssr-worker.js`
4. The test runner loads the worker and executes test cases against it

## Compatibility Flags

The test uses `nodejs_compat_v2` along with several additional Node.js module flags required
by the OpenNext runtime. These flags are configured in:

> - `src/wrangler.jsonc` - For the wrangler build (uses `compatibility_date: "2025-12-01"`)

Reviewer: The date here will get out of sync with the file, I know I tried ;)

Note that this compat date is only important for bundling the app with `open-next`. It will determine what set of `unenv` polyfills are used.

BUT the actual compat date used for running the tests is different.

I think that is an important detail to add because this will break the tests at some point (when we require a minimum compat date)

```suggestion
- `src/wrangler.jsonc` - For the wrangler build
```

- `opennext-ssr-test.wd-test` - For the workerd test runtime

Key flags include:

> - `nodejs_compat_v2` - Enables Node.js compatibility layer

Reviewer: aha...
Could you add a comment for that

Also I think I added a comment on

```capnp
  enableNodeJsFsModule @105 :Bool
    $compatEnableFlag("enable_nodejs_fs_module")
    $compatDisableFlag("disable_nodejs_fs_module")
    $impliedByAfterDate(name = "nodeJsCompat", date = "2025-09-15");
  # Enables the Node.js fs module. It is required to use this flag with
  # nodejs_compat (or nodejs_compat_v2).
  ```
  
  Saying that we should probably remove "(or nodejs_compat_v2)" here
  
  Do we need to do that and add `nodejs_compat` here too?

- `enable_nodejs_os_module` - Required for `node:os` module
- `enable_nodejs_fs_module` - Required for `node:fs` module
- `enable_nodejs_vm_module` - Required for `node:vm` module
- `enable_nodejs_http_modules` - Required for `node:http` modules

## Why JavaScript Instead of TypeScript?

The Next.js app uses JavaScript (`.js`/`.jsx`) instead of TypeScript because Next.js
automatically modifies `tsconfig.json` during builds. This conflicts with Bazel's sandbox,

> which treats source files as read-only. Using JavaScript with `jsconfig.json` avoids this issue.

Reviewer 2: Is this point still applicable if the sandbox is disabled?
Edit: Is it because one is for nextj.s and the other is for opennextjs-build?
